packages/legend-query-builder/src/stores/QueryBuilderResultState.ts

```diff
@@ -113,12 +142,69 @@ export class QueryBuilderResultState {
     this.previewLimit = Math.max(1, val);
   };
 
+  addCellData = (val: CellData): void => {
+    this.selectedCells.push(val);
+  };
+
+  setSelectedCells = (val: CellData[]): void => {
+    this.selectedCells = val;
+  };
+
+  setMouseOverCell = (val: CellData | null): void => {
+    this.mousedOverCell = val;
+  };
+
   setQueryRunPromise = (
     promise: Promise<ExecutionResult> | undefined,
   ): void => {
     this.queryRunPromise = promise;
   };
 
+  findColumnFromCoordinates = (
+    colIndex: number,
+  ): string | number | boolean | null | undefined => {
+    if (
+      !this.executionResult ||
+      !(this.executionResult instanceof TDSExecutionResult)
+    ) {
+      return undefined;
+    }
+    return this.executionResult.result.columns[colIndex];
+  };
+
+  findRowFromRowIndex = (
```

Comment:
should we make this one return undefined instead if we get unexpected results or out of bound?